test/parallel/rolldown.withdraw.merkleRoot.test.ts

```diff
@@ -0,0 +1,472 @@
+    expect(event.range.from.toNumber()).toEqual(nextRequestId.Ethereum);
+    expect(event.range.to.toNumber()).toEqual(
+      nextRequestId.Ethereum + (batchSize - 1),
+    );
+  });
+
+  test("Given a withdraw extrinsic run WHEN tokens are available THEN a withdraw will happen and l2Requests will be stored waiting for batch", async () => {
+    await signTx(
+      getApi(),
+      await Withdraw(testUser, 10, gaspIdL1Asset.ethereum, "Ethereum"),
+      testUser.keyRingPair,
+    ).then((events) => {
+      const res = getEventResultFromMangataTx(events);
+      expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    });
+    const event = await Rolldown.waitForNextBatchCreated(
+      "Ethereum",
+      waitingBatchPeriod,
+    );
+    const l2Request = await Rolldown.getL2Request(event.range.to.toNumber());
+    expect(l2Request.withdrawal.tokenAddress).toEqual(gaspIdL1Asset.ethereum);
+    expect(l2Request.withdrawal.withdrawalRecipient).toEqual(
+      testUser.keyRingPair.address,
+    );
+  });
+
+  test("Given a withdraw extrinsic run WHEN token is Eth and user has some tokens THEN a withdraw will happen and update will be stored on l2Requests and will be stored waiting for batch", async () => {
+    await Sudo.batchAsSudoFinalized(
+      Assets.mintToken(ETH_ASSET_ID, testUser, BN_MILLION),
+    );
+    await signTx(
+      getApi(),
+      await Withdraw(testUser, BN_HUNDRED, ethIdL1Asset.ethereum, "Ethereum"),
+      testUser.keyRingPair,
+    ).then((events) => {
+      const res = getEventResultFromMangataTx(events);
+      expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    });
+    const event = await Rolldown.waitForNextBatchCreated(
+      "Ethereum",
+      waitingBatchPeriod,
+    );
+    const l2Request = await Rolldown.getL2Request(event.range.to.toNumber());
+    expect(l2Request.withdrawal.tokenAddress).toEqual(ethIdL1Asset.ethereum);
+    expect(l2Request.withdrawal.withdrawalRecipient).toEqual(
+      testUser.keyRingPair.address,
+    );
+  });
+
+  test("Given a withdraw in a batch extrinsic, When one of the calls fail from the batch, then tokens are reverted ( transactional )", async () => {
+    const nonExistedToken = await getNonExistedTokenId();
+    const events = await Sudo.batchAsSudoFinalized(
+      ...(await getWithdrawalExtrinsics(10)),
+      await Withdraw(
+        testUser,
+        10,
+        nonExistedToken.toHuman().ethereum,
+        "Ethereum",
+      ),
+    );
+    const error = events.filter(
+      (x) => x.method === "ExtrinsicFailed" && x.section === "system",
+    );
+    expect(error.length).toBe(1);
+    expect(error[0].error!.name).toBe("TokenDoesNotExist");
+    await testUser.refreshAmounts(AssetWallet.AFTER);
+    const withdrawalAmount = testUser
+      .getAsset(ETH_ASSET_ID)!
+      .amountBefore.free.sub(testUser.getAsset(ETH_ASSET_ID)!.amountAfter.free);
+    expect(withdrawalAmount).bnEqual(BN_ZERO);
```

Comment:
Would it be possible to add an L2Updates storage check?

Reply:
Added